src/libostree/ostree-core.c

```diff
@@ -302,6 +303,47 @@ ostree_validate_collection_id (const char *collection_id, GError **error)
   return TRUE;
 }
 
+static int
+compare_xattrs (const void *a_pp, const void *b_pp)
+{
+  GVariant *a = *(GVariant **)a_pp;
+  GVariant *b = *(GVariant **)b_pp;
+  const char *name_a;
+  const char *name_b;
+  g_variant_get (a, "(^&ay@ay)", &name_a, NULL);
```

Comment:
Random: you could just `memcmp()` the data of the GVariants directly: the key starts at byte 0.  That's pretty evil, though.

Otherwise it might be a bit nicer to use `g_variant_get_child(a, 0, "&ay", &name_a)`